### PR TITLE
Add role update functionality to acl token update

### PR DESCRIPTION
**Pull request by Kartik1397**
This pull request adds the ability to update roles in the `acl token update` command.

Fixes: #18354

### CODE REVIEW

File: command/acl_token_update.go [hunk 1]
```diff
@@ -122,6 +141,10 @@ func (c *ACLTokenUpdateCommand) Run(args []string) int {
 		token.Policies = policies
 	}
 
+	if len(c.roleNames) != 0 || len(c.roleIDs) != 0 {
+		token.Roles = generateACLTokenRoleLinks(c.roleNames, c.roleIDs)
+	}
```
Comment (lgfa29) [inline command/acl_token_update.go:146]: One thing this makes me think is that these flags are additive, meaning they only allow adding new roles to the token, but there's no way to remove them.

Perhaps we also need something like `-unset-role-name` and `-unset-role-id`, similar to [`node meta -uset`](https://developer.hashicorp.com/nomad/docs/commands/node/meta/apply#unset). that would remove the given roles from the token?

File: command/acl_token_update.go [hunk 2]
```diff
@@ -38,6 +41,12 @@ Update Options:
   -policy=""
     Specifies a policy to associate with the token. Can be specified multiple times,
     but only with client type tokens.
+
+  -role-id=""
+     ID of a role to use for this token. May be specified multiple times.
+
+  -role-name=""
+     Name of a role to use for this token. May be specified multiple times.
```
Comment (lgfa29) [inline command/acl_token_update.go:49]:
```suggestion
  -role-id=""
    ID of a role to use for this token. Can be specified multiple times, but
    only with client type tokens.

  -role-name=""
    Name of a role to use for this token. Can be specified multiple times, but
    only with client type tokens.
```

Match the help message from `-policy`. I'm pretty sure the client type restriction also applies.

**Review by lgfa29 — CHANGES_REQUESTED** [review]
Looking good @Kartik1397, thank you for the PR! I left a suggestion about how to handle the removal of roles from a token.

Would you also be able to update the documentation page to include these new flags?
https://github.com/hashicorp/nomad/blob/main/website/content/docs/commands/acl/token/update.mdx

And also run the command `make cl` to generate a CHANGELOG entry.

Thanks again for the contribution!

File: command/acl_token_update.go (hunk 1 above)
Comment (tgross) [reply command/acl_token_update.go:140]: The `-role-id` and `-role-name` flags are like the `-policy` flags: they completely replace the existing polices or roles if set. I've updated the documentation to make that clear.

**Review by tgross — APPROVED** [review]
I've rebased this PR on `main`, pushed a changelog entry, resolved the open comments, and added documentation updates. Will merge once this is green and that'll ship in Nomad 1.8.3.

**Pull request merged**